src/pip/_internal/operations/build/metadata_editable.py

```diff
@@ -38,4 +38,5 @@ def generate_editable_metadata(
             except InstallationSubprocessError as error:
                 raise MetadataGenerationFailed(package_details=details) from error
 
+    assert distinfo_dir is not None
```

Comment:
I don't think this should be necessary given that `prepare_metadata_for_build_editable()` on the super class is typed to return `str`. This seems an oversight when type annotations were added to `pyproject-hooks`. I've asked around whether this is intentional.

Reply:
See also https://github.com/pypa/pyproject-hooks/pull/209 where I propose fixing `prepare_metadata_for_build_editable()`'s return type to `str`.